tests/expressions/test_parser.py

```diff
@@ -199,3 +199,7 @@ def test_with_function() -> None:
         parser.parse("foo = 1 and lower(bar) = '2'")
 
     assert "Expected end of text, found 'and'" in str(exc_info)
+
+
+def test_nested_field_equality() -> None:
+    assert EqualTo("foo.first", "a") == parser.parse("foo.first == 'a'")
```

Comment:
I think the key to success is to have some kind of syntax for quoting literals. For example: https://spark.apache.org/docs/latest/sql-ref-literals.html

Then we can parse something like:
```python
'a.b' -> Reference(('a.b',))
'a.b'.c -> Reference(('a.b', 'c'))
a.b.c -> Reference(('a', 'b', 'c'))
```

Or folks have to use:
```python
row_filter=EqualTo(('a.b',), 123)
row_filter=EqualTo(('a.b', 'c'), 123)
row_filter=EqualTo(('a', 'b', 'c'), 123)
```

Reply:
There is also an interesting proposal on the spec side of things: https://github.com/apache/iceberg/pull/10883/